metadata-ingestion/src/datahub/ingestion/source/sql/two_tier_sql_source.py

```diff
@@ -118,16 +118,23 @@ def get_inspectors(self):
         engine = create_engine(url, **self.config.options)
         with engine.connect() as conn:
             inspector = inspect(conn)
+            if inspector is None:
+                return None
```

Comment:
why was this required?

Reply:
This is a None check. There are scenarios where after successfully connecting to the source it is unable to run queries on the schema. This usually happens when the READ access is missing for a schema for hive source.
Adding this check ignores the schema and continues to the next one.

Sample exception

`sqlalchemy.exc.OperationalError: (pyhive.exc.OperationalError) TExecuteStatementResp(status=TStatus(statusCode=3, infoMessages=['*org.apache.hive.service.cli.HiveSQLException:Error while compiling statement: FAILED: SemanticException MetaException(message:java.security.AccessControlException: Permission denied: user=svc_vdc_ana_opr_prod, access=READ, inode="/projects/cdp/data/cda_map/tables":svchdpr221p:Hadoop_ope_cda:drwxr-x---\n\tat org.apache.hadoop.hdfs.server.namenode.FSPermissionChecker.check(FSPermissionChecker.java:409)\n\tat`

Reply:
@hsheth2 
Also, there is a scenario where Show Databases returns a Database/schema name but when it tries to connect it fail saying Database does not exist.

`sqlalchemy.exc.OperationalError: (pyhive.exc.OperationalError) TExecuteStatementResp(status=TStatus(statusCode=3, infoMessages=['*org.apache.hive.service.cli.HiveSQLException:Error while compiling statement: FAILED: SemanticException [Error 10072]: Database does not exist: None:28:27', 'org.apache.hive.service.cli.operation.Operation:toSQLException:Operation.java:342', 'org.apache.hive.service.cli.operation.SQLOperation:prepare:SQLOperation.java:200', 'org.apache.hive.service.cli.operation.SQLOperation:runInternal:SQLOperation.java:263'`

Reply:
I have reverted the code. The current code fails for the above 2 scenarios. What would be the right place to handle these errors?

Reply:
is the create_engine call causing the exception? or something else?

Reply:
Seems so, after adding the try catch block ingestion was continuing with the next schema/database.
Unable to reproduce as the source data is corrected now